src/directives/colony/clearRoom.ts

```diff
@@ -52,7 +60,9 @@ export class DirectiveClearRoom extends Directive {
 			for (let s of allStructures) {
 				if (s.structureType == STRUCTURE_CONTROLLER) continue;
 				if (keepStorageStructures &&
-					(s.structureType == STRUCTURE_STORAGE || s.structureType == STRUCTURE_TERMINAL)) {
+					(s.structureType == STRUCTURE_STORAGE || s.structureType == STRUCTURE_TERMINAL) && hasContents(s.store)) {
+					// Create a collection flag
+					let result = s.pos.createFlag(undefined, DirectiveHaul.color, DirectiveHaul.secondaryColor);
```

Comment:
sigh... I'm gonna need to work out all the bugs in DirectiveHaul, aren't I 😆

Reply:
Heh, it's definitely a useful one. I fixed some of them roughly in the power mining PR.